test/compile.jl

```diff
@@ -71,6 +72,16 @@ try
     end
     Base.compilecache("FooBar")
     sleep(2)
+    @test isfile(joinpath(dir, "FooBar.ji"))
+
+    touch(FooBar_file)
+    insert!(Base.LOAD_CACHE_PATH, 1, dir2)
+    Base.recompile_stale(:FooBar, joinpath(dir, "FooBar.ji"))
+    sleep(2)
```

Comment:
I hope this is enough for slow or highly-loaded systems? Would it be better to do a polling file watcher or something similar?